iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/import_/Importers.java

```diff
@@ -383,9 +389,31 @@ public static Importer findImporter(ReadOnlyDataSource dataSource, ComputationMa
         return findImporter(dataSource, LOADER.get(), computationManager, CONFIG.get());
     }
 
+    public static Importer findImporter(ReadOnlyDataStore dataStore, String fileName, ImportersLoader loader, ComputationManager computationManager, ImportConfig config) {
+        for (Importer importer : Importers.list(loader, computationManager, config)) {
+            if (importer.exists(dataStore, fileName)) {
+                return importer;
+            }
+        }
+        return null;
+    }
+
     public static Network loadNetwork(Path file, ComputationManager computationManager, ImportConfig config, Properties parameters, ImportersLoader loader) {
+        ReadOnlyDataStore dataStore;
+
+        try {
+            dataStore = DataStores.createDataStore(file);
+        } catch (IOException nde) {
+            throw new PowsyblException("Invalid file.");
+        }
+        String fileName = Files.isDirectory(file) ? null : file.toString();
```

Comment:
What is expected when we import a directory? Do we really want to support it?

Reply:
We will try to resolve a unique datapack, if it is not possible it will throw a NonUniqueResultException